src/main/java/hudson/plugins/ec2/ssh/EC2UnixLauncher.java

```diff
@@ -366,6 +366,13 @@ private Connection connectToSsh(EC2Computer computer, TaskListener listener) thr
                 }
                 String host = getEC2HostAddress(computer);
 
+                if ((computer.getNode() instanceof EC2SpotSlave) && computer.getInstanceId() == null) {
```

Comment:
```suggestion
                if ((node instanceof EC2SpotSlave) && computer.getInstanceId() == null) {
```
`getNode()` is called above